src/strategyHelpers/canary/canaryHelper.ts

```diff
@@ -213,6 +217,7 @@ async function cleanUpCanary(
       for (const inputObject of parsedYaml) {
          const name = inputObject.metadata.name
          const kind = inputObject.kind
+         const ns = inputObject?.metadata?.namespace
```

Comment:
💡 Thought to share:  with this `optional changing` what happens with the scenario, we supply `undefined` in the end? or we might want to add some error handlers in place?

Reply:
Added some additional typing to make it more explicit.